File: Course_1-Algorithmic_Toolbox/Week-1/Excercise_Challenges/2_maximum_pairwise_product/max_pairwise_product.cpp
#include <iostream>
#include <vector>
typedef long long ll;
using namespace std;

ll MaxPairwiseProduct(vector<int>& num)
{
    int max1 = -999, max2 = -9999, n = num.size();

    for(int i = 0; i < n; ++i){
        if(max1 < num[i]){
            max2 = max1;
            max1 = num[i];
        }
        else if(max2 < num[i]){
            max2 = num[i];
        }
    }

    return (ll)max1 * max2;
}

int main(void)
{
    int n;
    cin >> n;

    vector<int> v(n);
    for(int i = 0; i < n; ++i)
        cin >> v[i];

    cout << MaxPairwiseProduct(v) << "\n";

    return 0;
}

File: Course_1-Algorithmic_Toolbox/Week-1/Excercise_Challenges/2_maximum_pairwise_product/max_pairwise_product.py
def max_pairwise_product(numbers):
    max1 = -999
    max2 = -9999
    for value in numbers:
        if value > max1:
            max2 = max1
            max1 = value
        elif value > max2:
            max2 = value

    return max1 * max2


if __name__ == '__main__':
    input_n = int(input())
    input_numbers = [int(x) for x in input().split()]
    print(max_pairwise_product(input_numbers))
